domain/src/Commands/DomainCommands.php

```diff
@@ -0,0 +1,1288 @@
+   * Reassign old_domain entities, of the supplied type, to the new_domain.
+   *
+   * @param string $entity_type
+   *   The entity type name, e.g. 'node'
+   * @param string $field
+   *   The field to manipulate in the entity, e.g. DOMAIN_ACCESS_FIELD.
+   *   @todo: should this really be a string[] of fields?
+   * @param \Drupal\domain\DomainInterface $old_domain
+   *   The domain the entities currently belong to. It is not an error for
+   *   entity ids to be passed in that are not in this domain, though of course
+   *   not very useful.
+   * @param \Drupal\domain\DomainInterface $new_domain
+   *   The domain the entities should now belong to: When an entity belongs to
+   *   the old_domain, this domain replaces it.
+   * @param array $ids
+   *   List of entity IDs for the selected domain and all of type $entity_type.
+   *
+   * @return int
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  protected function reassignEntities(string $entity_type, string $field, DomainInterface $old_domain, DomainInterface $new_domain, $ids) {
+    $entity_storage = \Drupal::entityTypeManager()->getStorage($entity_type);
+
+    // @TODO is there a problem loading many, possibly big, entities in one go?
+    $entities = $entity_storage->loadMultiple($ids);
+
+    // @TODO do we need to protect against $entity not having the field $field?
+    /** @var \Drupal\domain\DomainInterface $entity */
+    foreach($entities as $entity) {
+      $changed = FALSE;
+      // Multivalue fields are used, so check each one.
+      foreach ($entity->get($field) as $k => $item) {
+        // NB: NOT strict ==
+        if ($item->target_id == $old_domain->id()) {
+
+          if ($this->is_dry_run) {
+            $this->logger()->info(dt('DRYRUN: Update domain membership for entity @id to @new.',
+              [ '@id' => $entity->id(), '@new' => $new_domain->id() ]));
+
+            // Don't set changed, so don't save either.
+            continue;
+          }
+
+          $changed = TRUE;
+          $item->target_id = $new_domain->id();
+        }
+      }
+      if ($changed) {
+        $entity->save();
+      }
+    }
+    return count($entities);
+  }
+
+  /**
+   * Return the Domain object corresponding to a policy string.
+   *
+   * @param string $policy
+   *   In general one of 'prompt' | 'default' | 'ignore' or a domain entity
+   *   machine name, but this function does not process 'prompt'.
+   *
+   * @return \Drupal\Core\Entity\EntityInterface|\Drupal\domain\DomainInterface|null
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  protected function getDomainInstanceFromPolicy(string $policy) {
+    $domain_storage = $this->getStorage();
+    switch($policy) {
+      case 'default':
+        $new_domain = $domain_storage->loadDefaultDomain();
+        break;
+
+      case 'prompt':
+      case 'ignore':
+        return NULL;
+
+      default:
```

Comment:
This gets confusing since we have a `case 'default'` above, suggesting that we need to renamed the policy option `default` to something else?

Reply:
I have updated the comments around here: hopefully this resolves it.